src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1452,9 +1456,36 @@ else if (item instanceof ItemMonsterPlacer)
         return modId;
     }
 
-    public static boolean onFarmlandTrample(World world, BlockPos pos, IBlockState state, float fallDistance, Entity entity)
+    private static final Map<EntityLiving.SpawnPlacementType, BiPredicate<IBlockAccess, BlockPos>> customSpawnPlacements = new IdentityHashMap<>();
+
+    static
     {
+        customSpawnPlacements.put(null, null);
+        customSpawnPlacements.put(EntityLiving.SpawnPlacementType.ON_GROUND, null);
+        customSpawnPlacements.put(EntityLiving.SpawnPlacementType.IN_AIR, null);
+        customSpawnPlacements.put(EntityLiving.SpawnPlacementType.IN_WATER, null);
+    }
 
+    public static void registerPlacementType(EntityLiving.SpawnPlacementType type, BiPredicate<IBlockAccess, BlockPos> predicate)
```

Comment:
Do not put modder facing API in ForgeHooks. Also this doesn't prevent overriding. Probably better to re-write the Enum class to accept a new test function as a argument.